onert-micro/eval-driver/Driver.cpp

```diff
@@ -0,0 +1,186 @@
+  }
+
+  const char *filename = argv[1];
+  const int32_t num_inputs = atoi(argv[2]);
+  const char *input_prefix = argv[3];
+  const char *output_file = argv[4];
+
+  std::ifstream file(filename, std::ios::binary | std::ios::in);
+  if (!file.good())
+  {
+    std::string errmsg = "Failed to open file";
+    throw std::runtime_error(errmsg.c_str());
+  }
+
+  file.unsetf(std::ios::skipws);
+
+  file.seekg(0, std::ios::end);
+  auto fileSize = file.tellg();
+  file.seekg(0, std::ios::beg);
+
+  // reserve capacity
+  DataBuffer model_data(fileSize);
+
+  // read the data
+  file.read(model_data.data(), fileSize);
+  if (file.fail())
+  {
+    std::string errmsg = "Failed to read file";
+    throw std::runtime_error(errmsg.c_str());
+  }
+
+  // Create interpreter.
+  luci_interpreter::Interpreter interpreter(model_data.data());
+
+  // Set input.
+  // Data for n'th input is read from ${input_prefix}n
+  // (ex: Add.circle.input0, Add.circle.input1 ..)
+  const auto input_tensors = interpreter.getInputTensors();
+  assert(num_inputs == input_tensors.size());
+  for (int32_t i = 0; i < num_inputs; i++)
+  {
+    auto *input_tensor = input_tensors[i];
+    std::vector<char> input_data(getTensorSize(input_tensor));
+    readDataFromFile(std::string(input_prefix) + std::to_string(i), input_data.data(),
+                     input_data.size());
+    luci_interpreter::Interpreter::writeInputTensor(input_tensor, input_data.data(),
+                                                    input_data.size());
+  }
+
+  // Do inference.
+  interpreter.interpret();
+
+  // Get output.
+  const auto output_tensors = interpreter.getOutputTensors();
+  for (int i = 0; i < output_tensors.size(); i++)
+  {
+    const auto *output_tensor = output_tensors[i];
+    std::vector<char> output_data(getTensorSize(output_tensor));
+    luci_interpreter::Interpreter::readOutputTensor(output_tensor, output_data.data(),
+                                                    output_data.size());
+
+    // Output data is written in ${output_file}
+    // (ex: Add.circle.output0)
+    // Output shape is written in ${output_file}.shape
+    // (ex: Add.circle.output0.shape)
+    writeDataToFile(std::string(output_file) + std::to_string(i), output_data.data(),
+                    output_data.size());
+    // In case of Tensor output is Scalar value.
+    // The output tensor with rank 0 is treated as a scalar with shape (1)
+    if (output_tensor->shape().num_dims() == 0)
+    {
+      writeDataToFile(std::string(output_file) + std::to_string(i) + ".shape", "1", 1);
+    }
+    else
+    {
+      auto shape_str = std::to_string(output_tensor->shape().dim(0));
+      for (int j = 1; j < output_tensor->shape().num_dims(); j++)
+      {
+        shape_str += ",";
+        shape_str += std::to_string(output_tensor->shape().dim(j));
+      }
+      writeDataToFile(std::string(output_file) + std::to_string(i) + ".shape", shape_str.c_str(),
+                      shape_str.size());
```

Comment:
maybe something like this?

```suggestion
      const auto tensor_shape_file = std::string(output_file) + std::to_string(i) + ".shape";
      writeDataToFile(tensor_shape_file, shape_str.c_str(), shape_str.size());
```